discord/utils.py

```diff
@@ -573,64 +588,88 @@ def get(iterable: Iterable[T], **attrs: Any) -> T | None:
     return None
 
 
-async def get_or_fetch(obj, attr: str, id: int, *, default: Any = MISSING) -> Any:
-    """|coro|
+_FETCHABLE = TypeVar(
+    "_FETCHABLE",
+    bound="VoiceChannel | TextChannel | ForumChannel | StageChannel | CategoryChannel | Thread | Member | User | Guild | GuildEmoji",
+)
 
-    Attempts to get an attribute from the object in cache. If it fails, it will attempt to fetch it.
-    If the fetch also fails, an error will be raised.
+
+async def get_or_fetch(
+    obj: Guild | Client,
+    object_type: type[_FETCHABLE],
```

Comment:
Please retain compatibility with the older version and trigger a deprecation warning if old parameters are used